napari_skimage_regionprops/_regionprops.py

```diff
@@ -184,3 +196,371 @@ def ellipsoid_axis_lengths(table):
+        return_summary_statistics: bool = True,
+        size: bool = True,
+        intensity: bool = True,
+        perimeter: bool = False,
+        shape: bool = False,
+        position: bool = False,
+        moments: bool = False,
+        napari_viewer: Viewer = None) -> "pandas.DataFrame":
+    """
+    Add a table widget to a napari viewer with mapped summary statistics.
+
+    Adds a table widget to a given napari viewer with summary statistics that
+    relates objects of one reference label image to objects in other label
+    images. Matched intensity images should be given for intensity
+    measurements. If no intensity images are given, calculates only properties
+    not related to intensities. If a single label image is given, it executes
+    regular 'regionprops_table' function.
+    """
+    multichannel = True
+    suffixes = []
+    if napari_viewer is not None:
+        # store list of labels layers for saving results later
+        labels_layer_list = [None]*len(label_images)
+        for layer in napari_viewer.layers:
+            if type(layer) is napari.layers.Labels:
+                # Store in the same order as label_images
+                for i, labels in enumerate(label_images):
+                    if np.array_equal(layer.data, labels):
+                        labels_layer_list[i] = layer
+                        suffixes += [layer.name]
+                    # Store reference channel index
+                    if np.array_equal(reference_label_image, labels):
+                        ref_channel = i
+    # If single label image is provided, indicate to do single channel regionprops
+    # ref_channel is used as a flag to indicate single or multi-channel
+    if len(label_images) == 1:
+        ref_channel = None
+        multichannel = False
+        label_images = label_images[0]
+    # If no intensity image provided, indicate no intensity measurements
+    if len(intensity_images) == 0:
+        intensity_images = None
+    else:
+        intensity_images = np.asarray(intensity_images)
+
+    table_list = regionprops_map_channels_table(
+        labels_array=np.asarray(label_images),
+        intensity_image=intensity_images,
+        ref_channel=ref_channel,
+        intersection_area_over_object_area=intersection_area_over_object_area,
+        summary=return_summary_statistics,
+        suffixes=suffixes,
+        size=size,
+        intensity=intensity,
+        perimeter=perimeter,
+        shape=shape,
+        position=position,
+        moments=moments)
+
+    if napari_viewer is not None:
+        # turn table into a widget
+        from ._table import add_table
+        # Clear labels layer properties (avoid appending when re-running)
+        if multichannel:
+            channel = ref_channel
+        else:
+            channel = 0
+        labels_layer_list[channel].properties = {}
+        
+        # Store results in the properties dictionary:
+        for table in table_list:
```

Comment:
As mentioned above, as a user I presume everything is measured in the context of the `reference_labels` and stored in this one particular layer.